eth/backend.go

```diff
@@ -404,9 +404,29 @@ func New(stack *node.Node, config *ethconfig.Config, logger log.Logger) (*Ethere
 	if casted, ok := backend.engine.(*ethash.Ethash); ok {
 		ethashApi = casted.APIs(nil)[1].Service.(*ethash.API)
 	}
-	atomic.StoreUint32(&backend.waitingForPOSHeaders, 0)
+	atomic.StoreUint32(&backend.waitingForBeaconChain, 0)
+	startAssembleFunc := func(timestamp uint64, random common.Hash, suggestedFeeRecipient common.Address) (types.Block, error) {
```

Comment:
We should not perform complete mining here, only to initiate it. Per the spec: "if payloadAttributes is not null and the client is not SYNCING, and MUST **begin** a payload build process building on top of forkchoiceState.headBlockHash".  And the build process should start concurrently and last for up to 12 seconds. See https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.5/src/engine/specification.md#payload-build-process

Currently `startAssembleFunc` confusingly not only starts the build process, but waits for it to finish.